toolchain/sem_ir/typed_insts.h

```diff
@@ -567,6 +567,17 @@ struct IntLiteral {
   IntId int_id;
 };
 
+struct IntType {
+  static constexpr auto Kind =
+      InstKind::IntType.Define<Parse::NodeId>("int_type");
+
+  TypeId type_id;
+  IntKind int_kind;
+  // TODO: Consider adding a more compact way of representing either a small
+  // unsigned integer bit width or an inst_id.
+  InstId bit_width_id;
```

Comment:
Why InstId instead of an IntId? (I think it's required to be a literal, in essence?)

Reply:
We support `Core.Int(N)` where `N` is a symbolic expression. That will eventually be important for things like `impl forall [N:! BigInt] Core.Int(N) as Add`, but isn't much use right now given the general lack of implementation of generics.